quipucords/quipucords/tests_feature_flags.py

```diff
@@ -0,0 +1,154 @@
+        assert feature_name in dict_with_test_flags
+        assert dict_with_test_flags[feature_name] is feature_value
+
+
+def test_when_value_cant_be_cast_to_int():
+    """Tests if function does not add or update key-values to default dict.
+
+    if value can`t be cast to int.
+    """
+    with mock.patch.dict(os.environ, (
+            {'QPC_FEATURE_OVERALL_STATUS': 'wrong'})):
+        dict_with_test_flags = get_feature_flags_from_env()
+        assert 'OVERALL_STATUS' in dict_with_test_flags
+        assert dict_with_test_flags['OVERALL_STATUS'] is False
+
+
+@pytest.mark.parametrize(
+    'env_name,env_value,feature_name',
+    (('QPC_FEATURE_TEST', '10', 'TEST'),
+     ('QPC_FEATURE_TEST1', '3', 'TEST1'),
+     ('QPC_FEATURE_TEST2', '2000', 'TEST2')
+     )
+)
+def test_when_int_is_not_valid_value_for_env(
+        env_name, env_value, feature_name):
+    """Tests if function does not add values that are not present in the.
+
+    dict for valid env variables.
+    """
+    with mock.patch.dict(os.environ, ({env_name: env_value})):
+        dict_with_test_flags = get_feature_flags_from_env()
+        assert feature_name not in dict_with_test_flags
+
+
+@pytest.mark.parametrize(
+    'env_name,env_value,feature_name',
+    (('TEST_QPC_FEATURE_', '1', 'TEST'),
+     ('TEST_QPC_FEATURE_', '1', 'TEST_'),
+     ('QPC_TEST1_FEATURE_', '0', 'TEST1'),
+     ('QPC_TEST1_FEATURE_', '0', '_TEST1'),
+     ('QPC_TEST1_FEATURE_', '0', 'TEST1_')
+     )
+)
+def test_function_only_adds_names_follow_standard(
+        env_name, env_value, feature_name):
+    """Tests if function only adds variables that start with QPC_FEATURE_."""
+    with mock.patch.dict(os.environ, ({env_name: env_value})):
+        dict_with_test_flags = get_feature_flags_from_env()
+        assert feature_name not in dict_with_test_flags
+
+
+@pytest.mark.parametrize(
+    'env_name,env_value,feature_name,feature_value',
+    (('qpc_feature_test', '1', 'TEST', True),
+     ('qpc_feature_TEST1', '0', 'TEST1', False),
+     ('QPC_feature_TEST2', '0', 'TEST2', False),
+     ('qpc_FEATURE_test3', '1', 'TEST3', True)
+     )
+)
+def test_if_function_is_not_case_sensitive(env_name, env_value, feature_name,
+                                           feature_value):
+    """Tests if function is not case-sensitive and env names are stored following.
+
+    the standard.
+    """
+    with mock.patch.dict(os.environ, ({env_name: env_value})):
+        dict_with_test_flags = get_feature_flags_from_env()
+        assert feature_name in dict_with_test_flags
+        assert dict_with_test_flags[feature_name] is feature_value
+
+
+@pytest.fixture
+def setup_feature_flag_instance_for_tests():
+    """Set up instance of FeatureFlag class for tests."""
+    with mock.patch.dict(os.environ, {'QPC_FEATURE_TEST': '1'}):
+        dict_with_test_flags = get_feature_flags_from_env()
+        feature_flag_instance = FeatureFlag(dict_with_test_flags)
```

Comment:
If you change `FeatureFlag.__init__` method you can replace this lines with

```suggestion
        feature_flag_instance = FeatureFlag()
```